### PR TITLE
commander: only process one vehicle_command/action_request per cycle

### PR DESCRIPTION
**Pull request by dagar**
- things like arming requests can be dependent on current nav state
that might requested by a previous command, but the state machine
transition will only happen after command processing

### CODE REVIEW

File: src/modules/commander/Commander.cpp [hunk 1]
```diff
@@ -3079,7 +3084,10 @@ Commander::run()
 
 		px4_indicate_external_reset_lockout(LockoutComponent::Commander, _armed.armed);
 
-		px4_usleep(COMMANDER_MONITORING_INTERVAL);
+		// sleep if there are no vehicle_commands or action_requests to process
+		if (!_vehicle_command_sub.updated() && !_action_request_sub.updated()) {
+			px4_usleep(COMMANDER_MONITORING_INTERVAL);
```
Comment (julianoes) [inline src/modules/commander/Commander.cpp:3089]: Ah nice.

**Review by julianoes — DISMISSED** [review]
Makes sense!

**Comment by dagar** [comment]
The UI LED timing is dependent on counts per loop, I'll change it to use time directly.

**Comment by dagar** [comment]
@julianoes I got rid of the loop counters and updated the UI LED code to use timestamps directly.  The timing comment was wrong ("this runs at around 20Hz"), so if you have opinions about LED blinking now is a good time...

**Comment by julianoes** [comment]
I don't have an opinion about that LED, I wasn't even aware of it.

**Review by julianoes — APPROVED** [review]

**Pull request merged**